src/main/kotlin/lotto/model/WinningStatistics.kt

```diff
@@ -0,0 +1,23 @@
+package lotto.model
+
+import kotlin.math.floor
+
+class WinningStatistics private constructor(
+    val winningStatistics: LinkedHashMap<WinningRank, Int>
+) {
+
+    fun calculateEarningRatio(paymentPrice: Int): Double {
+        val earnings = winningStatistics.keys.sumOf { it.prizeMoney * (winningStatistics[it] ?: 0) }
+        return floor(earnings / paymentPrice.toDouble() * 100.0) / 100
+    }
+
+    companion object {
+        fun from(winningRanks: List<WinningRank>): WinningStatistics {
+            val statistics = WinningRank.values().filter { it != WinningRank.NONE }
+                .reversed()
+                .associateWith { rank -> winningRanks.count { rank == it } }
+
+            return WinningStatistics(statistics as LinkedHashMap<WinningRank, Int>)
+        }
```

Comment:
stream을 극한으로 쓰면 아래와 같이 표현할 수도 있답니다.
enum 순서를 보장하려면 EnumMap도 하나의 방법입니다~! (성능도 좋아용)

```suggestion
        fun from(winningRanks: List<WinningRank>): WinningStatistics {
                return WinningRank.values().filter { it != WinningRank.NONE }
                .associateWith { rank -> winningRanks.count { rank == it } }
                .toMap(EnumMap(WinningRank::class.java))
                .let(::WinningStatistics)
        }
```

Reply:
오 enum map 이라는 개념도 있군요~ 알려주셔서 감사합니다 👍 _👍